pkg/subnet/local.go

```diff
@@ -146,19 +153,65 @@ func (d *SubnetDeployer) doDeploy(chain string, chain_genesis string) error {
 	ctx := binutils.GetAsyncContext()
 
 	ux.Logger.PrintToUser("VM ready. Trying to boot network...")
-	info, err := cli.Start(
+	loadSnapshotOpts := []client.OpOption{
+		client.WithPluginDir(pluginDir),
+		client.WithExecPath(avalancheGoBinPath),
+	}
+	loadSnapshotsInfo, err := cli.LoadSnapshot(
+		ctx,
+		constants.DefaultSnapshotName,
+		loadSnapshotOpts...,
+	)
+	if err != nil {
+		if !strings.Contains(err.Error(), "already bootstrapped") {
+			return fmt.Errorf("failed to start network :%s", err)
+		}
+		ux.Logger.PrintToUser("Network has already been booted. Wait until healthy...")
+	} else {
+		ux.Logger.PrintToUser("Booting Network. Wait until healthy...")
+	}
+
+	d.log.Debug(loadSnapshotsInfo.String())
+
+	subnetIDs, blockchainsInfo, _, err := d.WaitForHealthy(ctx, cli, d.healthCheckInterval)
+	if err != nil {
+		_ = binutils.KillgRPCServerProcess()
+		return fmt.Errorf("failed to query network health: %s", err)
+	}
+
+	// choose next subnetID
+	sort.Strings(subnetIDs)
+	subnetId := ""
+	if len(subnetIDs) > 0 {
```

Comment:
What if it's not > 0? `subnetId` is empty? But then why do all this? And if it's > 1? Why is it not a loop?

Reply:
if there is no set of preloaded subnet ids, uses the empty subnet id (and so a new subnet id will be created by ANR as before). in this case it is > 0 always with the bootstrap snapshot, except in unit tests.

Reply:
Ok this is clear now. 

For the records, I still don't like very much that we pre-create subnets. I think the user should always be in control. They should create a subnet, and if they save that to **their** snapshot, then they can reuse it. I think this is overcomplicating things, as we can see in fact here.